main.js

```diff
@@ -20,32 +21,48 @@ function closePeer() {
     }
 }
 
-var adapter = utils.Adapter({
-    name: 'yamaha',
+var adapter;
 
-    unload: function (callback) {
-        try {
-            //adapter.setState(adapter.namespace + '.Realtime.MAIN.PWR', 'unloaded');
-            if (y5) y5.close(true);
-            refreshTimer.clear();
-            closePeer();
-            setTimeout(callback, 1700);
-        } catch (e) {
-            callback();
-        }
-    },
-    stateChange: function (id, state) {
-        if (state) {
-            devices.setrawval(soef.ns.no(id), state.val);
-            if (!state.ack) yamaha.execCommand(id, state.val);
-        }
-    },
-    ready: function () {
-        devices.init(adapter, function (err) {
-            main();
-        });
+function startAdapter(options) {
+    if (!options) {
+        options = {};
     }
-});
+
+    Object.assign(options, {
+        name: 'yamaha',
+
+        unload: function (callback) {
+            try {
+                //adapter.setState(adapter.namespace + '.Realtime.MAIN.PWR', 'unloaded');
+                if (y5) y5.close(true);
+                refreshTimer.clear();
+                closePeer();
+                for (const key of Objects.keys(timeouts)) {
+                    if(timeouts[key]) {
+                        clearTimeout(timeouts[key]);
+                    }
+                }
+                timeouts.unload = setTimeout(callback, 1700);
```

Comment:
Whats the purpose of this? Is it because we want to have closePeer function finished? If so, it would be better to await it or do it in cb of closePeer - I have not taken a look inside the function.

Reply:
this is still code from soef. TBH I did not really check what is behind this. The timeout looks like it was found experimentally...

Reply:
without other setting in ip-package the max unload duration is expected with 500ms ... so way too long without setting it in io-package

Reply:
Code in unload seems to be synchronous anyway, really don't know why soef put a timeout in here... I would just remove it

Reply:
done.